docs/content/SegmentedControl.mdx

```diff
@@ -0,0 +1,190 @@
+  <SegmentedControl.Item leadingIcon={FileCodeIcon}>Raw</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={PeopleIcon}>Blame</SegmentedControl.Item>
+</SegmentedControl>
+```
+
+### Convert to a dropdown on smaller viewports
+
+```jsx live drafts
+<SegmentedControl aria-label="File view" asDropdown={{narrow: true, large: false}}>
+  <SegmentedControl.Item leadingIcon={EyeIcon}>Preview</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={FileCodeIcon}>Raw</SegmentedControl.Item>
+  <SegmentedControl.Item leadingIcon={PeopleIcon}>Blame</SegmentedControl.Item>
+</SegmentedControl>
+```
+
+### Fill width of parent
+
+```jsx live drafts
+<SegmentedControl block aria-label="File view">
+  <SegmentedControl.Item>Preview</SegmentedControl.Item>
+  <SegmentedControl.Item>Raw</SegmentedControl.Item>
+  <SegmentedControl.Item>Blame</SegmentedControl.Item>
+</SegmentedControl>
+```
+
+### In a loading state
+
+```jsx live drafts
+<SegmentedControl loading aria-label="File view">
+  <SegmentedControl.Item>Preview</SegmentedControl.Item>
+  <SegmentedControl.Item>Raw</SegmentedControl.Item>
+  <SegmentedControl.Item>Blame</SegmentedControl.Item>
+</SegmentedControl>
+```
+
+### With a label and caption on the left
+
+```jsx live drafts
+<Box display="flex">
+  <Box flexGrow={1}>
+    <Text fontSize={2} fontWeight="bold" id="scLabel-vert" display="block">
+      File view
+    </Text>
+    <Text color="fg.subtle" fontSize={1} id="scCaption-vert" display="block">
+      Change the way the file is viewed
+    </Text>
+  </Box>
+  <SegmentedControl aria-labelledby="scLabel-vert" aria-describedby="scCaption-vert">
+    <SegmentedControl.Item>Preview</SegmentedControl.Item>
+    <SegmentedControl.Item>Raw</SegmentedControl.Item>
+    <SegmentedControl.Item>Blame</SegmentedControl.Item>
+  </SegmentedControl>
+</Box>
+```
+
+### With a label above and caption below
+
+```jsx live drafts
+<FormControl>
+  <FormControl.Label id="scLabel-horiz">File view</FormControl.Label>
+  <SegmentedControl aria-labelledby="scLabel-horiz" aria-describedby="scCaption-horiz">
+    <SegmentedControl.Item>Preview</SegmentedControl.Item>
+    <SegmentedControl.Item>Raw</SegmentedControl.Item>
+    <SegmentedControl.Item>Blame</SegmentedControl.Item>
+  </SegmentedControl>
+  <FormControl.Caption id="scCaption-horiz">Change the way the file is viewed</FormControl.Caption>
+</FormControl>
+```
+
+### With something besides the first option selected
+
+```jsx live drafts
+<SegmentedControl aria-label="File view">
+  <SegmentedControl.Item>Preview</SegmentedControl.Item>
+  <SegmentedControl.Item selected>Raw</SegmentedControl.Item>
+  <SegmentedControl.Item>Blame</SegmentedControl.Item>
+</SegmentedControl>
+```
+
+### With a selection change handler
+
+```jsx live drafts
+<SegmentedControl
+  aria-label="File view"
+  onChange={selectedIndex => {
+    alert(`Segment ${selectedIndex}`)
+  }}
```

Comment:
Because the items are buttons, adding an onClick would also automatically work. Do we want to support both onClick on item and onChange on parent?

Reply:
I like the `onChange` on the parent so that we don't have to add repetitive `onClick` handlers to each `SegmentedControl.Button`